plugin/core/sessions.py

```diff
@@ -1089,11 +1089,18 @@ def run_code_action_async(self, code_action: Union[Command, CodeAction], progres
         code_action = cast(CodeAction, code_action)
         return self._maybe_resolve_code_action(code_action).then(self._apply_code_action_async)
 
-    def open_uri_async(self, uri: DocumentUri, r: RangeLsp, flags: int = 0, group: int = -1) -> Promise[bool]:
+    def open_uri_async(
+        self,
+        uri: DocumentUri,
+        r: Optional[RangeLsp],
+        flags: int = 0,
+        group: int = -1
+    ) -> Promise[bool]:
         if uri.startswith("file:"):
             # TODO: open_file_and_center_async seems broken for views that have *just* been opened via on_load
             path = self.config.map_server_uri_to_client_path(uri)
-            self.window.open_file(to_encoded_filename(path, r["start"]), flags | sublime.ENCODED_POSITION, group)
+            pos = r["start"] if r else {"line": 0, "character": 0}  # type: Position
+            self.window.open_file(to_encoded_filename(path, pos), flags | sublime.ENCODED_POSITION, group)
```

Comment:
I guess another option would be to not position at all when not provided.
Could make a different (for better or worse, not sure) when the file is already open.

Reply:
Yes, good idea for improvement